### PR TITLE
Add translation page

### PR DESCRIPTION
**Pull request by atch2203**
Will need to resolve conflicts on AppControllerComponent since I changed it for testing

### CODE REVIEW

**Review by rajkive — COMMENTED** [review]
Looks good to me.

File: frontend/src/components/AppControllerComponent/AppControllerComponent.js [hunk 1]
```diff
@@ -195,10 +195,11 @@ export class AppControllerComponent {
         case "exercise":
           viewContainer.appendChild(this.#exercisePageComponent.render());
           break;
-        case "translate":
+        case "translation":
           viewContainer.appendChild(this.#translationPageComponent.render());
+          break;
         default:
-          throw Error("Invalid View");
+          throw Error(`Invalid View ${this.#currentView}`);
```
Comment (chrispho) [inline frontend/src/components/AppControllerComponent/AppControllerComponent.js:202]: This is a nice change to add more information to the error

File: frontend/src/components/TranslationPageComponent/TranslationPageComponent.js [hunk 2]
```diff
@@ -22,55 +30,117 @@ export class TranslationPageComponent extends Component {
...
+    `; // TODO add inputs here
+  }
 
-    `;// TODO add inputs here
+  translate() {
+    const input = this.#inputElem.value;
+    const inputLang = this.#inputLangElem.value;
+    const outputLang = this.#outputLangElem.value;
+
+    EventHub.getInstance().publish(Events.Translate, {
```
Comment (chrispho) [inline frontend/src/components/TranslationPageComponent/TranslationPageComponent.js]: You can define the EventHub as a private variable to follow the convention of other components
`#hub = null`

and then put in the constructor
`this.#hub = EventHub.getInstance()`

and reference the private variable #hub as needed in your functions

**Review by chrispho — CHANGES_REQUESTED** [review]

File: frontend/src/components/AppControllerComponent/AppControllerComponent.js (hunk 1 above)
Comment (atch2203) [reply frontend/src/components/AppControllerComponent/AppControllerComponent.js:202]: ok

File: frontend/src/components/TranslationPageComponent/TranslationPageComponent.js (hunk 2 above)
Comment (atch2203) [reply frontend/src/components/TranslationPageComponent/TranslationPageComponent.js]: done

**Review by chrispho — APPROVED** [review]
lgtm

**Pull request merged**